test/features/dataset_deletion.feature

```diff
@@ -4,16 +4,16 @@ Feature: Dataset deletion
     Scenario: Sysadmin creates and deletes a dataset
         Given "SysAdmin" as the persona
         When I log in
-        And I create a dataset with name "dataset-deletion" and title "Dataset deletion"
+        And I create a dataset with key-value parameters "name=dataset-deletion::title=Dataset deletion"
         And I edit the "dataset-deletion" dataset
-        Then I press the element with xpath "//a[@data-module='confirm-action']"
+        When I press the element with xpath "//a[@data-module='confirm-action']"
         Then I should see "Briefly describe the reason for deleting this dataset"
         And I should see an element with xpath "//div[@class='modal-footer']//button[@class='btn btn-primary' and @disabled='disabled']"
         When I fill in "deletion-reason" with "it should be longer than 10 characters" if present
         Then I should not see an element with xpath "//div[@class='modal-footer']//button[@class='btn btn-primary' and @disabled='disabled']"
-        Then I press the element with xpath "//button[@class='btn btn-primary' and contains(text(), 'Confirm') ]"
+        When I press the element with xpath "//button[@class='btn btn-primary' and contains(text(), 'Confirm') ]"
         And I reload page every 2 seconds until I see an element with xpath "//div[contains(@class, "alert") and contains(text(), "Dataset has been deleted")]" but not more than 5 times
-        And I should not see "Dataset deletion"
+        Then I should not see "Dataset deletion"
         When I go to "/ckan-admin/trash"
         Then I should see "Dataset deletion"
-        Then I press the element with xpath "//form[contains(@id, 'form-purge-package')]//*[contains(text(), 'Purge')]"
+        When I press the element with xpath "//form[contains(@id, 'form-purge-package')]//*[contains(text(), 'Purge')]"
```

Comment:
I'm not sure if this should be a "When" or an "And". While it is an action and not a check, it's the last line in the test and doesn't have a corresponding "Then", and reading it as English it makes more sense to use "And", although it doesn't make a lot of difference either way.

Reply:
It will make a difference in future Behaving versions. The "I press the element with xpath" step will be annotated as `@when`, not `@step`, so if we try to associate it with "Then", we'll get a StepNotImplementedError.

It wouldn't be a bad idea to put an expectation after this, though.